web-app/src/pages/Dashboard/container.js

```diff
@@ -11,90 +11,88 @@ const DashboardContainer = Wrapped =>
+    };
+
+    // Filter Employees
+
+    filterEmployee = (filteredEvents, activeEmployee) => {
+      if (activeEmployee === -1) {
+        return filteredEvents;
       } else {
-        this.props.fetchEventsByUserId(id);
+        return filteredEvents.filter(hol => {
+          if (!hol.employee || hol.employee.employeeId == activeEmployee) {
+            return true;
+          }
+        });
       }
     };
 
-    onFilterEmployee = ({ employeeId }) => {
-      this.setState({ selectedEmployeeId: employeeId });
-      this.getTakenHolidaysById(parseInt(employeeId));
-    };
+    // Filter Events
 
-    updateFilterEvents = () => {
-      let eventKeys = [...this.state.eventKeysFilter];
-      let takenHolidaysUpdated = [];
-      if (eventKeys.length > 0) {
-        takenHolidaysUpdated = this.props.takenHolidays.filter(hol =>
-          eventKeys.includes(hol.eventStatus.eventStatusId)
+    filterEvents = (filteredEvents, activeEventIds) => {
+      if (activeEventIds.length === 0) {
+        return filteredEvents;
+      } else {
+        return filteredEvents.filter(hol =>
+          activeEventIds.includes(hol.eventStatus.eventStatusId)
         );
       }
+    };
 
-      this.setState({
-        takenHolidaysFiltered: takenHolidaysUpdated,
-      });
+    setActiveEvents = activeEventIds => {
+      this.setState({ activeEventIds }, this.filterCalenderEvents);
     };
 
-    onFilterEvents = eventStatusId => {
-      let eventKeys = [...this.state.eventKeysFilter];
-      if (eventStatusId !== undefined) {
-        if (eventKeys.includes(eventStatusId)) {
-          eventKeys = eventKeys.filter(item => item !== eventStatusId);
-        } else {
-          eventKeys.push(eventStatusId);
-        }
-      }
-      this.setState(
-        {
-          eventKeysFilter: eventKeys,
-        },
-        () => {
-          this.updateFilterEvents();
-        }
-      );
+    setActiveEmployee = employeeId => {
+      this.setState({ activeEmployee: employeeId }, this.filterCalenderEvents);
     };
 
     render() {
       return (
         this.props.userDetails && (
           <Wrapped
             employeeId={this.props.userDetails.employeeId}
-            takenHolidays={this.props.takenHolidays}
-            takenHolidaysFiltered={
-              this.state.takenHolidaysFiltered.length === 0
-                ? this.props.takenHolidays
-                : this.state.takenHolidaysFiltered
-            }
-            updateTakenHolidays={this.props.fetchEvents}
+            takenEvents={this.props.takenEvents}
+            events={this.state.filteredEvents}
+            updateTakenEvents={this.props.fetchEvents}
             isEventBeingUpdated={this.props.isEventBeingUpdated}
-            onUpdateEvents={this.onFilterEvents}
-            onUpdateEmployee={this.onFilterEmployee}
+            onUpdateEvents={activeEventIds =>
+              this.setActiveEvents(activeEventIds)
+            }
+            onUpdateEmployee={({ employeeId }) =>
```

Comment:
I might have confused myself here, but why is `{ employeeId }` destructured? can this not just be `onUpdateEmployee={employeeId =>`? apologies if I've got this wrong

Reply:
I think this gets passed an employee object